lib/transform.js

```diff
@@ -113,15 +113,29 @@ function getLinesToToc (lines, currentToc, info) {
   return lines.slice(tocableStart);
 }
 
-exports = module.exports = function transform(content, mode, maxHeaderLevel, title) {
+exports = module.exports = function transform(content, mode, maxHeaderLevel, title, notitle) {
   mode = mode || 'github.com';
   // only limit *HTML* headings by default
-  var maxHeaderLevelHtml = maxHeaderLevel || 4; 
-  title = title || '**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*';
+  var maxHeaderLevelHtml = maxHeaderLevel || 4;
+  var defaultTitle = '**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*';
 
   var lines = content.split('\n')
     , info = updateSection.parse(lines, matchesStart, matchesEnd)
 
+  if (notitle) {
+    title = '';
+  }
+  else if (title) {
+    // nop: title is already initialized
+  }
+  else if (info.hasStart) {
+    // Assume title is the second line after the start comment
+    title = lines[info.startIdx + 2];
+  }
+  else {
+    title = defaultTitle;
+  }
+
```

Comment:
This block is a little odd.
Couldn't we create a `determineTitle` function and then use early `return` to make this nicer?

```js
function determineTitle(title, notitle, defaultTitle, lines, info) {
  if (notitle) return '';
  if (title) return title;
  return info.hasStart ? lines[..] : defaultTitle; 
}
```

Something like that looks much more readable to me.